src/Components/ArticleCard.jsx

```diff
@@ -1,6 +1,25 @@
 import { Link } from "react-router";
+import { useState } from "react";
+import { updateArticleVotes } from "../api";
 
 const ArticleCard = ({ article }) => {
+  const [voteChange, setVoteChange] = useState(0);
+  const [hasVoted, setHasVoted] = useState(false);
+
+  const voteUpdater = (change) => {
+    if (!hasVoted) {
+      setVoteChange((prev) => prev + change);
+      updateArticleVotes(article.article_id, change)
+        .then(() => {
+          setHasVoted(true);
+        })
+        .catch((err) => {
+          setVoteChange((prev) => prev - change);
+          alert("Vote failed, please try again later.")
```

Comment:
Alerts are not the best way to handle this. They take the user away from what they're viewing and force an interaction to move on.